File: app/common/blogplanState.js
/** 探店计划状态，与 fs_dict.blogplan_state 一致 */
export const BLOGPLAN_STATE = {
  PENDING: 1,
  COMPLETED: 2,
  CANCELLED: 3
}

const LABEL_TO_CODE = {
  '待探店': BLOGPLAN_STATE.PENDING,
  '已完成': BLOGPLAN_STATE.COMPLETED,
  '已取消': BLOGPLAN_STATE.CANCELLED
}

const CODE_TO_LABEL = {}
Object.keys(LABEL_TO_CODE).forEach((k) => {
  CODE_TO_LABEL[LABEL_TO_CODE[k]] = k
})

export function blogplanStateCode(label) {
  if (label == null || label === '') return null
  const key = String(label).trim()
  return LABEL_TO_CODE[key] != null ? LABEL_TO_CODE[key] : null
}

export function blogplanStateLabel(code) {
  if (code == null) return ''
  return CODE_TO_LABEL[Number(code)] || ''
}

export function blogplanStateText(row) {
  if (!row) return ''
  if (row.statecn) return row.statecn
  const label = blogplanStateLabel(row.state)
  if (label) return label
  return typeof row.state === 'string' ? row.state : ''
}

File: src/main/java/com/ideabobo/constant/BlogplanState.java
package com.ideabobo.constant;

import java.util.Collections;
import java.util.HashMap;
import java.util.Map;

/** 探店计划状态 fs_dict.blogplan_state */
public final class BlogplanState {

    public static final int PENDING = 1;
    public static final int COMPLETED = 2;
    public static final int CANCELLED = 3;

    private static final Map<Integer, String> CODE_TO_LABEL;
    private static final Map<String, Integer> LABEL_TO_CODE;

    static {
        Map<Integer, String> c2l = new HashMap<>();
        c2l.put(PENDING, "待探店");
        c2l.put(COMPLETED, "已完成");
        c2l.put(CANCELLED, "已取消");
        CODE_TO_LABEL = Collections.unmodifiableMap(c2l);
        Map<String, Integer> l2c = new HashMap<>();
        for (Map.Entry<Integer, String> e : c2l.entrySet()) {
            l2c.put(e.getValue(), e.getKey());
        }
        LABEL_TO_CODE = Collections.unmodifiableMap(l2c);
    }

    private BlogplanState() {
    }

    public static String labelOf(Integer code) {
        if (code == null || code <= 0) {
            return "";
        }
        String label = CODE_TO_LABEL.get(code);
        return label != null ? label : "";
    }

    public static Integer codeOfLabel(String label) {
        if (label == null) {
            return null;
        }
        String t = label.trim();
        if (t.isEmpty()) {
            return null;
        }
        Integer exact = LABEL_TO_CODE.get(t);
        if (exact != null) {
            return exact;
        }
        if (t.contains("待探店")) {
            return PENDING;
        }
        if (t.contains("已完成") || t.contains("完成")) {
            return COMPLETED;
        }
        if (t.contains("已取消") || t.contains("取消")) {
            return CANCELLED;
        }
        return null;
    }
}
